### DISCUSSION TITLE
No animations with revealer

### DISCUSSION THREAD

**Discussion by 0x7375** [elkowar/eww#861] [Q&A] [answered]
I can't figure out how to make a revealer animate, it's always immediate.
Shouldn't this minimal config  work?

```
(defpoll clock_time :interval "1s" "date +%H:%M")
(defpoll clock_date :interval "1s" "date '+%d/%m'")

(defvar rev false)
(defvar eww "eww -c .")

(defwidget clock []
  (eventbox :onhover "${eww} update rev=true"
            :onhoverlost "${eww} update rev=false"
            :space-evenly "false" 
    (box 
         :style "font-weight: bold;"
         :space-evenly "false" 
         :spacing "20" 
         :halign "center"
      (label :text "${clock_time}")
      (revealer :transition "slideright" 
                :duration "500ms"
                :reveal rev
                (label :text "${clock_date}")
      )
    )
  )
)

(defwindow test
  :monitor 0
  :geometry (geometry :anchor "top center" :width "100%" :height "50px")
  :exclusive true
  (clock))
```
I'm using hyprland if that's relevant

**Comment by viandoxdev** [comment]
I'd suggest taking a look at the information in #196. Revealers tend to be a little annoying to work with, but since most of the problems come from GTK itself there isn't much that can be done on the ew side.

**Comment by 0x7375** [comment]
I read through this, but I didn't really understand. I installed Arch with Gnome before switching to Hyprland, is it possible that doing a fresh Hyprland install could fix this?

**Reply by viandoxdev** [reply]
Did you follow the steps over [there](https://github.com/elkowar/eww/issues/196#issuecomment-917643152) ? Also no the window manager / compositor most definitely doesn't affect revealers in themselves.

**Reply by 0x7375** [reply]
Yes, I copied the widget in a new empty folder and I still do not get any animation on show/hide

https://github.com/elkowar/eww/assets/91350939/8b62bec7-0677-412e-b450-03dd14375bb6

**Reply by viandoxdev** [reply]
Well sorry I can't help you more right now (I'm away from my computer until the 17th). I guess just start with the simplest config possible (even with a crossfade revealer) and  debug from there. Make sure you didn't toggle any weird settings in the GTK debugger (eww inspector).

**Reply by 0x7375** [reply, marked as answer]
Well, I did not know about eww inspector, I had "animations" off ......... I think I changed it in gnome (with dconf editor) to get rid of workspace animations or something like that, suggested by some forum probably. Thank you!
